File: NetBeansProjects/AnimalBoardGame/src/main/java/AnimalBoardGame/Dice.java
package AnimalBoardGame;

import java.util.Random;


public class Dice 
{
    private Random  rand;
    private int     firstRoll;
    private int     secondRoll;
    
    public Dice()
    {   
        rand = new Random();
        firstRoll = -1;
        secondRoll = -1;
    }

    public int GetPairOfDiceRoll()
    {
        firstRoll = GetSingleDiceRoll();
        secondRoll = GetSingleDiceRoll();
        
        return firstRoll + secondRoll;     
    }
    
    public boolean ThrewDouble()
    {
        if (firstRoll != -1 && secondRoll != -1)
        {
            return firstRoll == secondRoll;
        }
        else return false;                    
    }

    private int GetSingleDiceRoll()
    {
        return rand.nextInt(6) + 1;
    }
}

File: C#/Animalopoly/Dice.cs
using System;

namespace Animalopoly
{
    class Dice 
    {
        private int firstThrow;
        private int secondThrow;

        private Random rand;
    
        public Dice()
        {
            rand = new Random();
            firstThrow = -1;
            secondThrow = -1;
        }

        public int GetRollTotal()
        {
            firstThrow = SingleRoll();
            secondThrow = SingleRoll();

            return firstThrow + secondThrow;
        }
    
        public bool ThrewDouble()
        {
            if (firstThrow != -1 && secondThrow != -1)
            {
                return firstThrow == secondThrow;
            }
            else
                return false;        
        }
    
        private int SingleRoll()
        {
            return rand.Next(1,7);
        }
    }
}
